src/data-source.handlers.ts

```diff
@@ -0,0 +1,52 @@
+import { RequestHandler } from 'express';
+import { DataSourceModel } from './data-source.model';
+import {
+  normalizeHttpError,
+  NotFoundHttpError
+} from './lib/normalize-http-error';
+import uuidv4 from 'uuid/v4';
+import {dataSourcesPath} from "./data-source.router";
+
+interface ResourceHandlerMap {
+  [key: string]: RequestHandler;
+}
+
+export const dataSourceHandlers: ResourceHandlerMap = {
```

Comment:
Perhaps a class implementation would be cleaner?

Reply:
It used to be a class with inheritance. I second that it should still be a class (but with static handlers this time and no inheritance)